pkg/backup/item_collector.go

```diff
@@ -99,6 +100,64 @@ func (r *itemCollector) getGroupItems(log logrus.FieldLogger, group *metav1.APIR
 	return items, nil
 }
 
+// sortResourcesByOrder sorts items by the names specified in "order".  Items are not in order will be put at the end in original order.
+func sortResourcesByOrder(log logrus.FieldLogger, items []*kubernetesResource, order []string) []*kubernetesResource {
+	if len(order) == 0 {
+		return items
+	}
+	log.Debugf("Sorting resources using the following order %v...", order)
+	itemMap := make(map[string]*kubernetesResource)
+	for _, item := range items {
+		var fullname string
+		if item.namespace > "" {
+			fullname = fmt.Sprintf("%s/%s", item.namespace, item.name)
+		} else {
+			fullname = item.name
+		}
+		itemMap[fullname] = item
+	}
+	var sortedItems []*kubernetesResource
+	// First select items from the order
+	for _, name := range order {
```

Comment:
should we trim trailing spaces on `name`?
say the `velero backup create` command supplied an ordering of `pods=ns1/pod1, ns1/pod2; persistentvolumeclaims=ns1/pvc4, ns1/pvc8`
with a ` ` after the `,`

Reply:
I'll do "strings.TrimSpace"